test/integration/provider/admin/backend_apis_controller_test.rb

```diff
@@ -58,7 +58,7 @@ class Provider::Admin::BackendApisControllerTest < ActionDispatch::IntegrationTe
 
   test 'delete a backend api with products' do
     backend_api = @provider.backend_apis.order(:id).first
-    FactoryBot.create(:backend_api_config, service: @provider.first_service, backend_api: backend_api)
+    FactoryBot.create(:backend_api_config, service: FactoryBot.create(:simple_service, account: @provider), backend_api: backend_api)
```

Comment:
If you had to change to different the service here, it's because a `BackendApiConfig` already existed with `backend_api`. I think if you remove this line the test will pass just the same.